tests/common/mod.rs

```diff
@@ -104,7 +121,205 @@ pub async fn setup_bitcoind() -> BitcoindNode {
+
+                    retry = false;
+                    retry_num += 1;
+                }
+                Err(err) => {
+                    println!(
+                        "getting client error {err}, retrying call {} time",
+                        retry_num
+                    );
+                    if retry_num == 3 {
+                        panic!("could not set up client: {err}")
+                    }
+                    retry = true;
+                    retry_num += 1;
+                }
+            }
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn get_info(&mut self) -> tonic_lnd::lnrpc::GetInfoResponse {
+        let get_info_req = tonic_lnd::lnrpc::GetInfoRequest {};
+
+        let resp = if let Some(ref mut client) = &mut self.client {
+            let mut retry_num = 0;
+            let resp = Err(());
+            while retry_num < 3 {
+                thread::sleep(Duration::from_secs(2));
+                match client.lightning().get_info(get_info_req.clone()).await {
+                    Err(_) => {
+                        println!("retrying get_info call");
+                        retry_num += 1;
+                        if retry_num == 3 {
+                            panic!("get_info call failed after 3 retries");
+                        }
+                        continue;
+                    }
+                    Ok(resp) => return resp.into_inner(),
+                };
+            }
+            resp
+        } else {
+            panic!("Client is None");
+        };
+
+        resp.unwrap()
+    }
+
+    // connect_to_peer connects to the specified peer.
+    pub async fn connect_to_peer(
+        &mut self,
+        node_id: PublicKey,
+        addr: SocketAddr,
+    ) -> tonic_lnd::lnrpc::ConnectPeerResponse {
+        let ln_addr = tonic_lnd::lnrpc::LightningAddress {
+            pubkey: node_id.to_string(),
+            host: addr.to_string(),
+        };
+
+        let connect_req = tonic_lnd::lnrpc::ConnectPeerRequest {
+            addr: Some(ln_addr),
+            timeout: 20,
+            ..Default::default()
+        };
+
+        let resp = if let Some(ref mut client) = &mut self.client {
+            let mut retry_num = 0;
+            let resp = Err(());
+            while retry_num < 3 {
+                thread::sleep(Duration::from_secs(2));
+                match client.lightning().connect_peer(connect_req.clone()).await {
+                    Err(_) => {
+                        println!("retrying connect_peer call");
+                        retry_num += 1;
+                        if retry_num == 3 {
+                            panic!("connect_peer call failed after 3 retries");
+                        }
+                        continue;
+                    }
+                    Ok(resp) => return resp.into_inner(),
+                };
+            }
+            resp
+        } else {
+            panic!("Client is None");
+        };
```

Comment:
DRY: retry code can be pulled out for generic use

Reply:
good call